File: android/src/main/java/com/icapps/rnpushnotifications/RNPushNotificationsPackage.java
package com.icapps.rnpushnotifications;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.JavaScriptModule;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * @author Koen Van Looveren
 */
public class RNPushNotificationsPackage implements ReactPackage {
	@Override
	public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
		final List<NativeModule> list = new ArrayList<>();
		list.add(new RNPushNotificationsModule(reactContext));
		return list;
	}

	@Override
	public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
		return Collections.emptyList();
	}

	@Override
	@Deprecated()
	public List<Class<? extends JavaScriptModule>> createJSModules() {
		return Collections.emptyList();
	}
}

File: android/src/main/java/com/icapps/rnpushnotifications/RNPushNotificationsPackage.kt
package com.icapps.rnpushnotifications

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.JavaScriptModule
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

import java.util.ArrayList
import java.util.Collections

/**
 * @author Koen Van Looveren
 */
class RNPushNotificationsPackage : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        val list = ArrayList<NativeModule>()
        list.add(RNPushNotificationsModule(reactContext))
        return list
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return Collections.emptyList()
    }

    override fun createJSModules(): List<Class<out JavaScriptModule>> {
        return Collections.emptyList()
    }
}